### ISSUE TITLE
[bug]: Combobox should have similar focus to select and other inputs

### ISSUE THREAD

**Issue by jkinley** [shadcn-ui/ui#4806]
### Describe the bug

Currently when you focus on the combobox there is no focus ring. However, if you tab to the combobox (using focus-visible) there is a ring. The goal should be to make the combobox work like the select regarding focus rings.

The problem is that the trigger for the combobox is a button and buttons in shadcn don't have a ring on focus. The Select trigger is a button too,  but it has the following tailwind classes:

The Combobox example does not have the correct classes on the trigger (Button).

<img width="522" alt="image" src="https://github.com/user-attachments/assets/d509dd69-c1fe-475d-9d2f-f021190b2dfc">

The Select component has focus classes on the trigger (Button):

<img width="958" alt="image" src="https://github.com/user-attachments/assets/dee2baf8-02f7-48be-9f70-3d0e451ed02c">

When I add the tailwind focus classes to my combobox it seems to work ok, but not quite the experience as the Select.

```
<Popover open={open} onOpenChange={setOpen}>
        <PopoverTrigger asChild>
          <Button
            variant="outline"
            role="combobox"
            aria-expanded={open}
            className="w-[200px] justify-between ring-offset-background focus:outline-none focus:ring-2 focus:ring-ring focus:ring-offset-2"
          >
```

<img width="329" alt="image" src="https://github.com/user-attachments/assets/3f8f4b68-ca67-4be0-acf2-f2fedd321f32">

but maybe you want to have more integrated solution.

### Affected component/components

Combobox

### How to reproduce

1. Click on combobox, is no focus ring.
2. Tab to combobox, there is focus ring.
3. Click on select, there is focus ring.
4. Tab to select, there is focus ring.

### Codesandbox/StackBlitz link

https://ui.shadcn.com/docs/components/combobox

### System Info

```bash
Chrome latest
Shadcn latest
```

### Before submitting

- [X] I've made research efforts and searched the documentation
- [X] I've searched for existing issues

**Comment by orendecor** [comment]
Dig into it, it makes sense the ring is not appearing on the combobox on click as we directly shift the focus to the popup, whereas while using the keyboard the focus is not shifting up until you click enter (and the ring disappear as intended).

I'm not sure there is a bug here

**Comment by shadcn** [comment]
This issue has been automatically marked as stale due to one year of inactivity. It will be closed in 7 days unless there’s further input. If you believe this issue is still relevant, please leave a comment or provide updated details. Thank you. (This is an automated message)

**Comment by shadcn** [comment]
This issue has been automatically closed due to one year of inactivity. If you’re still experiencing a similar problem or have additional details to share, please open a new issue following our current issue template. Your updated report helps us investigate and address concerns more efficiently. Thank you for your understanding! (This is an automated message)
